### PR TITLE
feat: Migrate final gamut-system docs + functions to variance [WEB-1413] 

### PR DESCRIPTION
**Pull request by codecaaron**
### Overview

[Variance Migration Doc
](https://www.notion.so/codecademy/Variance-Migration-a675bbc31b584fec9dfff72c0364b194)

It done :) 

Finalizes the last of the variance migration pieces.

Related PR (Closed and feedback addressed here): https://github.com/Codecademy/client-modules/pull/1645

* Adds basic documentation for the variant and css documentation for variance API in place of original gamut-styles documention as a stopgap.
* Migrates all remaining uses of old variant API to the new one.
* Exports both `css` and `variant` as named exports from gamut-styles (as well as on `system.x`).
* Updates imports for new variant API where possible.
* Removes gamut-system badge from README
* Changes gamut-kit instructions to include variance and removes gamut-kit.
* Removes gamut-system dependencies from `gamut-kit` `gamut-styles` `gamut` `gamut-labs`.
* Updates Icon build step to omit gamut-system.

### PR Checklist

- [ ] Related to designs:
- [x] Related to JIRA ticket: WEB-1413
- [x] I have run this code to verify it works
- [ ] This PR includes unit tests for the code change

### CODE REVIEW

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-icons@4.14.2-alpha.3d0bdb.0<br/>@codecademy/gamut-illustrations@0.15.2-alpha.3d0bdb.0<br/>@codecademy/gamut-kit@0.3.114-alpha.3d0bdb.0<br/>@codecademy/gamut-labs@18.1.5-alpha.3d0bdb.0<br/>@codecademy/gamut-patterns@0.2.2-alpha.3d0bdb.0<br/>@codecademy/gamut-styles@10.1.2-alpha.3d0bdb.0<br/>@codecademy/gamut-tests@2.4.2-alpha.3d0bdb.0<br/>@codecademy/gamut@33.1.1-alpha.3d0bdb.0<br/>@codecademy/konamimojisplosion@0.2.33-alpha.3d0bdb.0<br/>@codecademy/markdown-overrides@0.5.45-alpha.3d0bdb.0<br/>@codecademy/styleguide@40.0.2-alpha.3d0bdb.0<br/>@codecademy/variance@0.12.1-alpha.3d0bdb.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60ad9c877b389f00d66bf328--gamut-preview.netlify.app>https://60ad9c877b389f00d66bf328--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60ad9c877b389f00d66bf328>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-icons@4.14.3-alpha.2ab7a3.0<br/>@codecademy/gamut-illustrations@0.16.2-alpha.2ab7a3.0<br/>@codecademy/gamut-kit@0.3.117-alpha.2ab7a3.0<br/>@codecademy/gamut-labs@18.1.8-alpha.2ab7a3.0<br/>@codecademy/gamut-patterns@0.2.3-alpha.2ab7a3.0<br/>@codecademy/gamut-styles@10.1.3-alpha.2ab7a3.0<br/>@codecademy/gamut-tests@2.4.3-alpha.2ab7a3.0<br/>@codecademy/gamut@33.1.4-alpha.2ab7a3.0<br/>@codecademy/konamimojisplosion@0.2.34-alpha.2ab7a3.0<br/>@codecademy/markdown-overrides@0.5.46-alpha.2ab7a3.0<br/>@codecademy/styleguide@40.0.5-alpha.2ab7a3.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60ada3f547a266135bd3019a--gamut-preview.netlify.app>https://60ada3f547a266135bd3019a--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60ada3f547a266135bd3019a>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-icons@4.14.3-alpha.4de391.0<br/>@codecademy/gamut-illustrations@0.16.2-alpha.4de391.0<br/>@codecademy/gamut-kit@0.3.117-alpha.4de391.0<br/>@codecademy/gamut-labs@18.1.8-alpha.4de391.0<br/>@codecademy/gamut-patterns@0.2.3-alpha.4de391.0<br/>@codecademy/gamut-styles@10.1.3-alpha.4de391.0<br/>@codecademy/gamut-tests@2.4.3-alpha.4de391.0<br/>@codecademy/gamut@33.1.4-alpha.4de391.0<br/>@codecademy/konamimojisplosion@0.2.34-alpha.4de391.0<br/>@codecademy/markdown-overrides@0.5.46-alpha.4de391.0<br/>@codecademy/styleguide@40.0.5-alpha.4de391.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60ada61b29ba14167369c545--gamut-preview.netlify.app>https://60ada61b29ba14167369c545--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60ada61b29ba14167369c545>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-icons@4.14.3-alpha.002323.0<br/>@codecademy/gamut-illustrations@0.16.2-alpha.002323.0<br/>@codecademy/gamut-kit@0.3.117-alpha.002323.0<br/>@codecademy/gamut-labs@18.1.8-alpha.002323.0<br/>@codecademy/gamut-patterns@0.2.3-alpha.002323.0<br/>@codecademy/gamut-styles@10.1.3-alpha.002323.0<br/>@codecademy/gamut-tests@2.4.3-alpha.002323.0<br/>@codecademy/gamut@33.1.4-alpha.002323.0<br/>@codecademy/konamimojisplosion@0.2.34-alpha.002323.0<br/>@codecademy/markdown-overrides@0.5.46-alpha.002323.0<br/>@codecademy/styleguide@40.0.5-alpha.002323.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60add065a96ee951bc865754--gamut-preview.netlify.app>https://60add065a96ee951bc865754--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60add065a96ee951bc865754>Deploy Logs</a></p>

File: packages/styleguide/stories/Foundations/System/Variants.stories.mdx [hunk 1]
```diff
@@ -11,93 +11,173 @@ import { Meta } from '@storybook/addon-docs/blocks';
...
 
-```tsx
-const colorVariant = variant({
-  default: 'primary',
-  variants: {
-    primary: { color: 'blue' },
-    secondary: { color: 'green' },
-  },
-});
+In addition to system props, both of these functions permit configurations with any valid CSS property that has not been overriden. These properties are typed to their CSSType defaults and will not accept values with the responsive syntax.
```
Comment (dreamwasp) [inline packages/styleguide/stories/Foundations/System/Variants.stories.mdx:63]:
```suggestion
In addition to system props, both of these functions permit configurations with any valid CSS property that has not been overridden. These properties are typed to their [CSSType](https://github.com/frenic/csstype#csstype) defaults and will not accept values with the responsive syntax.
```

just for context i think

**Review by dreamwasp — APPROVED** [review]
🙌

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-icons@4.14.3-alpha.7d92da.0<br/>@codecademy/gamut-illustrations@0.16.2-alpha.7d92da.0<br/>@codecademy/gamut-kit@0.3.117-alpha.7d92da.0<br/>@codecademy/gamut-labs@18.1.8-alpha.7d92da.0<br/>@codecademy/gamut-patterns@0.2.3-alpha.7d92da.0<br/>@codecademy/gamut-styles@10.1.3-alpha.7d92da.0<br/>@codecademy/gamut-tests@2.4.3-alpha.7d92da.0<br/>@codecademy/gamut@33.1.4-alpha.7d92da.0<br/>@codecademy/konamimojisplosion@0.2.34-alpha.7d92da.0<br/>@codecademy/markdown-overrides@0.5.46-alpha.7d92da.0<br/>@codecademy/styleguide@40.0.5-alpha.7d92da.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60ae4e151c16a60c2d3d30f9--gamut-preview.netlify.app>https://60ae4e151c16a60c2d3d30f9--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60ae4e151c16a60c2d3d30f9>Deploy Logs</a></p>

File: README.md [hunk 2]
```diff
@@ -49,9 +51,9 @@ We provide a single package to manage the versions of a few core dependencies: `
 
 - [![npm version](https://badge.fury.io/js/%40codecademy%2Fgamut-labs.svg)](https://badge.fury.io/js/%40codecademy%2Fgamut-labs)
 
-[`gamut-system`: Typescript CSS in JS utility library](/packages/gamut-system/README.md)
+[`variance`: Typescript CSS in JS utility library](/packages/variance/README.md)
```
Comment (JoshuaKGoldberg) [inline README.md:54]: Nit: just noticed...

```suggestion
[`variance`: TypeScript CSS in JS utility library](/packages/variance/README.md)
```

File: packages/styleguide/stories/Foundations/System/About.stories.mdx [hunk 3]
```diff
@@ -13,6 +13,6 @@ import { TableOfContents } from '~styleguide/blocks';
 
 Many of our components are written in our CSS in JS library, [Emotion](https://emotion.sh/). we've created a set of style utilities that we call **System Props** to improve reusability, predictability, and standardization across our design system.
 
-These functions are built with [`@codecademy/gamut-system`](https://github.com/Codecademy/client-modules/tree/main/packages/gamut-system) a library inspired by [`styled-system`](https://styled-system.com/) but with a focus on typesafety.
+These functions are built with [`@codecademy/variance`](https://github.com/Codecademy/client-modules/tree/main/packages/variance) a library inspired by [`styled-system`](https://styled-system.com/) but with a focus on typesafety.
```
Comment (JoshuaKGoldberg) [inline packages/styleguide/stories/Foundations/System/About.stories.mdx:16]:
```suggestion
These functions are built with [`@codecademy/variance`](https://github.com/Codecademy/client-modules/tree/main/packages/variance): a library inspired by [`styled-system`](https://styled-system.com/) but with a focus on typesafety.
```

File: packages/styleguide/stories/Foundations/System/About.stories.mdx [hunk 4]
```diff
@@ -13,6 +13,6 @@ import { TableOfContents } from '~styleguide/blocks';
...
```
Comment (JoshuaKGoldberg) [inline packages/styleguide/stories/Foundations/System/About.stories.mdx:17]: I think this page is the first one many people will see when they try to read up on variance / system props, but it doesn't have much info and doesn't explain what order to read things in. Do you think you could give a quick high-level overview of the major features here?

**Review by JoshuaKGoldberg — APPROVED** [review]
LGTM! I think there's more to be done on entry-level docs but that certainly doesn't block this PR.

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-icons@4.14.4-alpha.f04379.0<br/>@codecademy/gamut-illustrations@0.16.2-alpha.f04379.0<br/>@codecademy/gamut-kit@0.3.120-alpha.f04379.0<br/>@codecademy/gamut-labs@18.2.3-alpha.f04379.0<br/>@codecademy/gamut-patterns@0.2.3-alpha.f04379.0<br/>@codecademy/gamut-styles@10.1.3-alpha.f04379.0<br/>@codecademy/gamut-tests@2.4.3-alpha.f04379.0<br/>@codecademy/gamut@33.1.6-alpha.f04379.0<br/>@codecademy/konamimojisplosion@0.2.34-alpha.f04379.0<br/>@codecademy/markdown-overrides@0.5.46-alpha.f04379.0<br/>@codecademy/styleguide@40.0.8-alpha.f04379.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60ae972ce428f600b1d5d267--gamut-preview.netlify.app>https://60ae972ce428f600b1d5d267--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60ae972ce428f600b1d5d267>Deploy Logs</a></p>

**Pull request merged**